### PR TITLE
Winnowing fix

### PR DESCRIPTION
**Pull request by MbolotSuse**
Fix for rancher/rancher#38011

### CODE REVIEW

File: pkg/api/steve/supportconfigs/generator.go [hunk 1]
```diff
@@ -136,7 +136,8 @@ func (h *GeneratorHandler) generateSupportConfig() (io.Reader, error) {
 		}
 	}
 
-	return &buf, nil
+	err = tw.Close()
+	return &buf, err
```
Comment (KevinJoiner) [inline pkg/api/steve/supportconfigs/generator.go:140]: nit: 
```suggestion
	return &buf, tw.Close()
```

**Review by KevinJoiner — APPROVED** [review]
LGTM

File: pkg/api/steve/supportconfigs/generator.go (hunk 1 above)
Comment (MbolotSuse) [reply pkg/api/steve/supportconfigs/generator.go:140]: I did think about doing this and this was what I originally had. However, I did not like the idea of having a return which called another function and returned a pre-computed value. I'm ok with one function call serving as the return, but one call and one static value just felt off to me. Let me know if you disagree - definitely see the point of having less LOC.

Comment (KevinJoiner) [reply pkg/api/steve/supportconfigs/generator.go:140]: I think it's readable either way,  Which is the only argument to be made here. It's purely a style decision at this point, so I would leave it if you prefer this way. One argument being a static value, should be fine and is similar to what is done [in the bytes package](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/bytes/buffer.go;l=168).

**Review by maxsokolovsky — APPROVED** [review]

**Pull request merged**